sdk/samples/iot/hub/src/paho_iot_hub_sas_telemetry_example.c

```diff
@@ -82,89 +82,109 @@ static const char* telemetry_message_payloads[NUMBER_OF_MESSAGES] = {
   {
-    printf("%s\n", hide_value ? "***" : env);
-    az_span env_span = az_span_from_str(env);
-    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(env_span));
-    az_span_copy(buffer, env_span);
-    *out_value = az_span_slice(buffer, 0, az_span_size(env_span));
+    printf("Failed to read configuration from environment variables, return code %d\n", rc);
+    return rc;
   }
-  else if (default_value != NULL)
+
+  // Get the MQTT client id used for the MQTT connection
+  size_t client_id_length;
+  if (az_failed(
+          rc = az_iot_hub_client_get_client_id(
+              &client, mqtt_client_id, sizeof(mqtt_client_id), &client_id_length)))
   {
-    printf("%s\n", default_value);
-    az_span default_span = az_span_from_str(default_value);
-    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(default_span));
-    az_span_copy(buffer, default_span);
-    *out_value = az_span_slice(buffer, 0, az_span_size(default_span));
+    printf("Failed to get MQTT clientId, return code %d\n", rc);
+    return rc;
   }
-  else
+
+  if (az_failed(rc = generate_sas_key()))
   {
-    printf("(missing) Please set the %s environment variable.\n", env_name);
-    return AZ_ERROR_ARG;
+    printf("Failed to get SAS key, return code %d\n", rc);
+    return rc;
   }
 
-  return AZ_OK;
-}
+  // Create the Paho MQTT client
+  if ((rc = MQTTClient_create(
+           &mqtt_client, mqtt_endpoint, mqtt_client_id, MQTTCLIENT_PERSISTENCE_NONE, NULL))
+      != MQTTCLIENT_SUCCESS)
+  {
+    printf("Failed to create MQTT client, return code %d\n", rc);
+    return rc;
+  }
 
-// Create mqtt endpoint e.g: ssl//contoso.azure-devices.net:8883
-static az_result create_mqtt_endpoint(char* destination, int32_t destination_size, az_span iot_hub)
-{
-  int32_t iot_hub_length = (int32_t)strlen(iot_hub_hostname);
-  int32_t required_size = az_span_size(mqtt_url_prefix) + iot_hub_length
-      + az_span_size(mqtt_url_suffix) + (int32_t)sizeof(null_terminator);
+  // Connect to IoT Hub
+  if ((rc = connect_device()) != 0)
+  {
+    return rc;
+  }
 
-  if (required_size > destination_size)
+  // Loop and send 5 messages
+  if ((rc = send_telemetry_messages()) != 0)
   {
-    return AZ_ERROR_INSUFFICIENT_SPAN_SIZE;
+    return rc;
   }
 
-  az_span destination_span = az_span_init((uint8_t*)destination, destination_size);
-  az_span remainder = az_span_copy(destination_span, mqtt_url_prefix);
-  remainder = az_span_copy(remainder, az_span_slice(iot_hub, 0, iot_hub_length));
-  remainder = az_span_copy(remainder, mqtt_url_suffix);
-  az_span_copy_u8(remainder, null_terminator);
+  printf("Messages Sent [Press ENTER to shut down]\n");
+  (void)getchar();
 
-  return AZ_OK;
+  // Gracefully disconnect: send the disconnect packet and close the socket
+  if ((rc = MQTTClient_disconnect(mqtt_client, TIMEOUT_MQTT_DISCONNECT_MS)) != MQTTCLIENT_SUCCESS)
+  {
+    printf("Failed to disconnect MQTT client, return code %d\n", rc);
+    return rc;
+  }
+  printf("Disconnected.\n");
```

Comment:
some messages has a trailing period, some do not. Most do not.

```suggestion
  printf("Disconnected\n");
```